packages/main/src/RPA/Excel/Files.py

```diff
@@ -787,6 +787,14 @@ def append_worksheet(self, name=None, content=None, header=False, start=None):
 
     def remove_worksheet(self, name=None):
         name = self._get_sheetname(name)
+        others = [sheet for sheet in self.sheetnames if sheet != name]
+
+        if not others:
+            raise ValueError("Workbook must have at least one worksheet")
```

Comment:
Just curious - why there needs to be this kind of protection? I mean why user can't just delete all sheets and create then new sheet.

Reply:
At least .xls workbooks (with the upstream library) can not have zero worksheets, it breaks in spectacular ways. With .xlsx that can be done, as long as the user adds one before saving. Not sure if I want the keywords to behave differently based on the file extension.

Reply:
Yeah that is fine then to keep the protection. Thanks 👍